csm/core/services/cluster_management.py

```diff
@@ -13,27 +13,64 @@
 # For any questions about this software or licensing,
 # please email opensource@seagate.com or cortx-questions@seagate.com.
 
+import time
 from csm.common.services import ApplicationService
 from cortx.utils.conf_store.conf_store import Conf
 from csm.core.blogic import const
 from cortx.utils.log import Log
-
+from csm.common.comm import MessageBusComm
+from csm.common.errors import CsmServiceNotAvailable
 
 class ClusterManagementAppService(ApplicationService):
     """
     Manage operations on cluster and resources in it.
     """
 
-    def __init__(self, plugin, message_bus_obj):
+    def __init__(self, plugin):
         self._cluster_management_plugin = plugin
-        self.message_bus_obj = message_bus_obj
-        self.message_bus_obj.init(type=const.PRODUCER,
-                            producer_id=Conf.get(const.CSM_GLOBAL_INDEX,
-                                                const.MSG_BUS_CLUSTER_STOP_PRODUCER_ID),
-                            message_type=Conf.get(const.CSM_GLOBAL_INDEX,
-                                                const.MSG_BUS_CLUSTER_STOP_MSG_TYPE),
-                            method=Conf.get(const.CSM_GLOBAL_INDEX,
-                                                const.MSG_BUS_CLUSTER_STOP_METHOD))
+        self.message_bus_obj = None
+
+    def init_message_bus_producer(self):
+        Log.debug("Initialising message bus Producer")
+        try:
+            self.message_bus_obj.init(type=const.PRODUCER,
+                    producer_id=Conf.get(const.CSM_GLOBAL_INDEX,
+                                        const.MSG_BUS_CLUSTER_STOP_PRODUCER_ID),
+                    message_type=Conf.get(const.CSM_GLOBAL_INDEX,
+                                        const.MSG_BUS_CLUSTER_STOP_MSG_TYPE),
+                    method=Conf.get(const.CSM_GLOBAL_INDEX,
+                                        const.MSG_BUS_CLUSTER_STOP_METHOD))
+        except Exception as e:
+            Log.error(f"Message bus failing: {e}")
+            return False
+        Log.debug("Message bus Producer initialized")
+        return True
+
+    def init_message_bus(self):
+        Log.debug("Initialising message bus communication")
+        self.message_bus_obj = MessageBusComm(Conf.get(const.CONSUMER_INDEX, const.KAFKA_ENDPOINTS),
+                                         unblock_consumer=True)
+        MAX_RETRY_COUNT = int(Conf.get(const.CSM_GLOBAL_INDEX, const.MAX_RETRY_COUNT))
+        RETRY_SLEEP_DURATION = int(Conf.get(const.CSM_GLOBAL_INDEX, const.RETRY_SLEEP_DURATION))
+
+        if self.message_bus_obj is None:
+            Log.error("Message bus communication is not available.")
+            raise CsmServiceNotAvailable()
```

Comment:
Change this to internal error and send proper message

Reply:
updated